package/scripts/postinstall

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# verify the files exist
+tmp_brew="$2"
+if [[ ! -d "${tmp_brew:?}" ]]; then
+    echo "no directory at $tmp_brew, exiting"
+    exit 1
+fi
+
+# pick the correct target
+if [[ "$3" != "/" ]]; then
+  target=$3
+elif [[ $(uname -m) == "x86_64" ]];then
+  target="/usr/local"
+else
+  target="/opt/Homebrew"
+fi
+
+loggedInUser=$(echo "show State:/Users/ConsoleUser" | scutil | awk '/Name :/ { print $3 }')
+if [[ -f "$target/bin/brew" ]]; then
+    if [[ $(sudo -u"$loggedInUser" "$target/bin/brew" developer) =~ "enabled" ]]; then
+      echo "developer install, exiting"
+      exit 0
+    fi
+    if [[ $("$tmp_brew/bin/brew" --version | head -n1) != $("$target/bin/brew" --version | head -n1) ]]; then
+      echo "already an outdated install at $target, updating"
+      sudo -u"$loggedInUser" "$target/bin/brew" update --auto-update
```

Comment:
What breaks under these scenarios that means overwriting with the pkg files is bad?

I wonder if it's worth just blocking reinstallation with a `installation-check` script with a message that teaches users about `brew update (&& brew upgrade)` instead.

Reply:
I don't think we can overwrite. And if we do, it'll get rid of all your taps.

Reply:
It does feel weird to have the installer be a no-op in the `brew developer` case (well, apart from leaving behind the tmp directory but that's probably an oversight) without any messaging.

I reckon if people are running the package for a second time then they are either wanting to reinstall Homebrew, which is not what happens and we should tell them how to do it properly.

Reply:
Or they're updating. In my personal scenario for this I'll have an MDM deploy Homebrew to developer machines and I'll enforce updates through the MDM.

Reply:
Given this won't allow updating to specific versions included in the `pkg`, I don't see any point of the `brew developer` check and I feel this section would make more sense without it.